sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -237,6 +237,11 @@ object GpuOrcScan extends Arm {
           case BOOLEAN | BYTE | SHORT | INT | LONG => true
           case _ => false
         }
+      case VARCHAR =>
```

Comment:
NIT:
It can also cover the CHAR to STRING path conveniently by

```suggestion
      case VARCHAR | CHAR =>
```
. [Since cuDF already supports this conversion too](https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/src/io/orc/reader_impl.cu#L84).

Reply:
I tried this, but it doesn't match the CPU behavior.  One of the main differences between VARCHAR and CHAR is that the latter strips trailing whitespace, so if the value of a CHAR(5) field is `xyz ` with a single trailing space then libcudf will pad it to the full character width as `xyz  ` with two trailing spaces but the CPU will load it as `xyz` with no trailing spaces.  I'll file a separate tracking issue, since that means we're not handling CHAR loads even without trying to cast to a different read schema like string.

Reply:
We're covered here in the sense that we won't silently load a corrupted CHAR value since it will always try to upcast to a string during load.  Added a comment to #6149 detailing the needed CHAR behavior.